006_ext-curl-set_opt-crash.patch
PHP_5_2
http://cvs.php.net/viewvc.cgi/php-src/ext/curl/interface.c?r1=1.62.2.14.2.42&r2=1.62.2.14.2.43&diff_format=u
http://bugs.php.net/bug.php?id=47616
Fixed bug #47616 (curl keeps crashing)
diff -r 9db107056020 -r 85ef4c1844f0 ext/curl/interface.c
--- a/ext/curl/interface.c	Fri Apr 10 11:25:01 2009 +0200
+++ b/ext/curl/interface.c	Fri Apr 10 11:29:41 2009 +0200
@@ -1550,6 +1550,7 @@
 			} else {
 #if LIBCURL_VERSION_NUM >= 0x071101
 				/* with curl 7.17.0 and later, we can use COPYPOSTFIELDS, but we have to provide size before */
+				convert_to_string_ex(zvalue);
 				error = curl_easy_setopt(ch->cp, CURLOPT_POSTFIELDSIZE, Z_STRLEN_PP(zvalue));
 				error = curl_easy_setopt(ch->cp, CURLOPT_COPYPOSTFIELDS, Z_STRVAL_PP(zvalue));
 #else
